modules/publish/Components.php

```diff
@@ -132,4 +134,162 @@ public function componentSpecialWhatLinksHere()
             $this->articles = Articles::getTable()->getAllByLinksToArticleName($this->linked_article_name);
         }
 
+        public function componentSpecialContributions()
+        {
+            $request = framework\Context::getRequest();
+            $current_project = framework\Context::getCurrentProject();
+
+            $available_page_sizes = array(20, 50, 100, 250, 500);
+            $default_page_size = 50;
+            $default_page = 1;
+
+            $username = $request->getParameter('user');
+            $page_size = ceil($request->getParameter('page_size', $default_page_size));
+            $page = floor($request->getParameter('page', $default_page));
+
+            // All contributions made by the user will be stored within this
+            // array.
+            $contributions = array();
+
+            // Fetch user (if any was passed-in).
+            $user = Users::getTable()->getByUsername($username);
+
+            // Calculate base URL and base URL user prefix. Prefix is used to
+            // avoid extra ? or & signs).
+            $base_url = make_url('publish_article', array('article_name' => "Special:{$this->projectnamespace}Contributions"));
+            $base_url_user_prefix = ($user !== null ? "?user={$username}&" : '?');
+
+            // Fetch history based on user.
+            $history = ArticleHistory::getTable()->getByUser($user);
+
+            if ($history)
+            {
+                // Contributions by the user will be stored in this specific array.
+                $contributions = array();
+
+                while ($row = $history->getNextRow())
+                {
+                    // Extract basic information.
+                    $date = $row[ArticleHistory::DATE];
+                    $revision = $row[ArticleHistory::REVISION];
+                    $article_name = $row[ArticleHistory::ARTICLE_NAME];
+                    $reason = $row[ArticleHistory::REASON];
+                    $author_id = $row[ArticleHistory::AUTHOR];
+                    $author = Users::getTable()->getByUserID($author_id);
+                    if ($author === null)
+                    {
+                        $author_contributions_url = null;
+                    }
+                    else
+                    {
+                        $author_contributions_url = "{$base_url}?user={$author->getUsername()}";
+                    }
+
+                    // Ignore articles the currently logged-in user can't read.
+                    $article = Article::getByName($article_name);
+                    if (!$article->canRead() || $article->getProject() != $current_project)
```

Comment:
Just realised that the canRead() will also check if project is archived. The second part of check (compare article's project against current project) is still necessary.